xde/src/xde.rs

```diff
@@ -962,19 +963,20 @@ fn clear_xde_underlay() -> Result<NoResp, OpteError> {
         };
```

Comment:
Comment about this whole block: What are we supposed to do if any of these unexpected conditions result in an error being emitted, since we're altering state as part of the attempted tear-down.  `take()`-ing the `underlay` for example.  If we emit `EBUSY`, then we won't have a path back to re-`take()`-ing it, right?

Reply:
Some of these errors are suggestive of programming errors, but we could do worse than to check all preconditions at the first instance and then proceed (the conditions we've setup, `no ports && write lock held`, should ensure nothing can violate those expectations). I'll revisit the logic here in the morning.